<?xml version="1.0" encoding="utf-8"?>

<launch>
  <!-- Console launch prefix -->
  <arg name="pkg_name" default="nepi_ai_yolo5"/>
  <arg name="node_name" default="yolov5_ros"/>
  <arg name="node_file" default="nepi_ai_yolov5_node.py"/>
  <arg name="launch_prefix" default=""/>
  <arg name="node_base_namespace" default="/nepi/" />

  <!-- Config and weights folder. -->
  <arg name="weights_path"          default="/mnt/nepi_storage/ai_models/yolov5_ros/models/weights"/>
  <arg name="config_path"           default="/mnt/nepi_storage/ai_models/yolov5_ros/models/cfg"/>

  <!-- network parameter files -->
  <arg name="network_param_file"         default="/mnt/nepi_storage/ai_models/yolov5_ros/config/common_object_detection_small.yaml"/>

  <!-- Input image -->
  <arg name="source_img_topic"                  default="$(arg node_base_namespace)/image_raw"/>

  <!-- Detection threshold -->
  <arg name="detection_threshold"        default="0.3"/>

  <!-- Load parameters -->
  <rosparam command="load" ns="$(arg node_base_namespace)/$(arg node_name)" file="$(arg network_param_file)"/>

  <!-- Start yolov5 and ros wrapper -->
  <node pkg="$(arg pkg_name)" ns="$(arg node_base_namespace)" type="$(arg node_file)" name="$(arg node_name)" output="screen" launch-prefix="$(arg launch_prefix)">
    <param name="node_base_namespace"   value="$(arg node_base_namespace)" />
    <param name="weights_path"          value="$(arg weights_path)" />
    <param name="config_path"           value="$(arg config_path)" />
    <param name="source_img_topic" value="$(arg source_img_topic)" />
    <param name="detection_threshold" value="$(arg detection_threshold)" />
  </node>

</launch>

